### ISSUE TITLE
[@typescript-eslint/typedef] objectDestructuring=false apparently not working

### ISSUE THREAD

**Issue by octogonz** [typescript-eslint/typescript-eslint#788]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/typedef": [
      "error",
      {
        "objectDestructuring": false,
        "variableDeclaration": true
      },
    ],
  }
}
```

```TS
interface Vector { x: number, y: number };
const v: Vector = { x: 1, y: 2 };

// error @typescript-eslint/typedef: expected a type annotation
const { x, y } = v;
```

**Expected Result**

It seems like maybe this error should not be reported, because `const { x, y } = v` is a destructuring initializer and we said `"objectDestructuring": false`. 

**Actual Result**

The error is reported.

**Additional Information**

I am not sure whether this by design, or a bug.  The lint error can be eliminated like this:
```ts
const { x, y }: Vector = v;
```

But will that work with a partial assignment such as `const { x }: Vector = v;`?  

Philosophically, although I like for things to be typed, I think of the destructuring variable initializer as a shorthand that does not introduce any new unknowns, and thus perhaps should not require extra annotations.  In my example we are using `x` as a shorthand for `v.x`. Presumably the lint policy has already ensured that we know the type of `v.x`.

@JoshuaKGoldberg  Do you have an opinion?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0-alpha.4` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |

**Comment by bradzacher** [comment]
There are test cases which are checking for this, that are passing in master:

https://github.com/typescript-eslint/typescript-eslint/blob/73f8c791e82aee0d5d8f5f8a7a6287126e597769/packages/eslint-plugin/tests/rules/typedef.test.ts#L80-L103

https://github.com/typescript-eslint/typescript-eslint/blob/73f8c791e82aee0d5d8f5f8a7a6287126e597769/packages/eslint-plugin/tests/rules/typedef.test.ts#L223-L248

So I'm not sure what the problem might be? Is something overriding your config?

**Comment by octogonz** [comment]
The test cases are missing `"variableDeclaration": true`.
